File: MATLAB/DecryptVigenere_German.m
function y = DecryptVigenere_German(x, k)
k1 = double(k) - 97;
n = size(x, 2);
m = size(k, 2);
for i = 1 : n
    xp = double(x(i)) - 65;
    ii = mod(i - 1, m) + 1;
    y(i) = mod(xp - k1(ii), 30) + 97;
end;
y = char(y);

File: JavaScript/DecryptVigenere_German.js
import { mod } from "./mod.js";

export function DecryptVigenere_German(ciphertext_ascii_30) {
    // Scramble using keyword
    var keyword = "bratwurst";
    var keyword_ascii = keyword.split('').map(function (char) { return char.charCodeAt(0) - 97; });
    for (var i = 0; i < ciphertext_ascii_30.length; i++) {
        ciphertext_ascii_30[i] = mod((ciphertext_ascii_30[i] - 65) - keyword_ascii[mod(i, keyword.length)], 30) + 97
    }
    return ciphertext_ascii_30
};
